pkg/controller/remoteingress/remoteingress_controller.go

```diff
@@ -233,5 +354,98 @@ func createIngressController(cd *hivev1.ClusterDeployment, ingress hivev1.Cluste
+
+	for _, ingress := range rContext.clusterDeployment.Spec.Ingress {
+		if ingress.ServingCertificate == "" {
+			continue
+		}
+
+		foundCertBundle := false
+		for _, cb := range rContext.clusterDeployment.Spec.CertificateBundles {
+			if cb.Name == ingress.ServingCertificate {
+				foundCertBundle = true
+				cbSecret := &corev1.Secret{}
+				searchKey := types.NamespacedName{
+					Name:      cb.SecretRef.Name,
+					Namespace: rContext.clusterDeployment.Namespace,
+				}
+
+				if err := r.Get(context.TODO(), searchKey, cbSecret); err != nil {
+					if errors.IsNotFound(err) {
+						msg := fmt.Sprintf("secret %v for certbundle %v was not found", cb.SecretRef.Name, cb.Name)
+						rContext.logger.Error(msg)
+						return cbSecrets, fmt.Errorf(msg)
+					}
+					rContext.logger.WithError(err).Error("error while gathering certBundle secret")
+					return cbSecrets, err
+				}
+
+				// gather the contents of the secret
+				// TODO: validate that the secret really contains TLS data
+				cbSecrets[cb.Name] = cbSecret.Data
+			}
+		}
+		if !foundCertBundle {
+			return cbSecrets, fmt.Errorf("didn't find expected certbundle %v", ingress.ServingCertificate)
+		}
+	}
+
+	return cbSecrets, nil
+}
+
+// setIngressCertificateNotFoundCondition will set/unset the condition indicating whether all certificates required
+// by the clusterDeployment ingress objects were found. Returns any error encountered while setting the condition.
+func (r *ReconcileRemoteClusterIngress) setIngressCertificateNotFoundCondition(rContext *reconcileContext, notFound bool, missingSecretMessage string) error {
+	var (
+		msg, reason string
+		status      corev1.ConditionStatus
+		updateCheck utils.UpdateConditionCheck
+	)
+
+	origCD := rContext.clusterDeployment.DeepCopy()
+
+	if notFound {
+		msg = missingSecretMessage
+		status = corev1.ConditionTrue
+		reason = ingressCertificateNotFoundReason
+		updateCheck = utils.UpdateConditionIfReasonOrMessageChange
+	} else {
+		msg = fmt.Sprintf("all secrets for ingress found")
+		status = corev1.ConditionFalse
+		reason = ingressCertificateFoundReason
+		updateCheck = utils.UpdateConditionNever
+	}
+
+	rContext.clusterDeployment.Status.Conditions = utils.SetClusterDeploymentCondition(rContext.clusterDeployment.Status.Conditions,
+		hivev1.IngressCertificateNotFoundCondition, status, reason, msg, updateCheck)
+
+	if !reflect.DeepEqual(rContext.clusterDeployment.Status.Conditions, origCD.Status.Conditions) {
+		if err := r.Status().Update(context.TODO(), rContext.clusterDeployment); err != nil {
+			rContext.logger.WithError(err).Error("error updating clusterDeployment condition")
+			return err
+		}
+	}
+
+	return nil
+}
+
+// secretNameForCertificateBundle just stitches together a secret name consisting of
+// the clusterDeployment.Name plus the name of the certificateBundle.Name
+func secretNameForCertificateBundle(certBundleName string, cd *hivev1.ClusterDeployment) string {
+	return fmt.Sprintf("%s-%s", cd.Name, certBundleName)
```

Comment:
I would think we want to create a secret with the same name that was specified in the certificate bundle

Reply:
the thinking i had with this naming scheme was to avoid someone naming the certbundle 'default' (or the secret referenced by the certBundle) and then trying to sync down a secret named 'default' on the remote cluster and getting a collision (not that there's a secret named 'default' in the 'openshift-ingress' namespace...but I hope i've explained the idea).

i'm open to just passing through the secret name, but what do you think about the collisions?

Reply:
I was thinking of being able to find the secret on the target cluster if I need to debug the target cluster directly. I know what I named my secret locally, but there's nothing telling me that it's now named something else on the target. I do agree that there's a danger of collisions, so I'm split :) 
If you think this is better, then let's keep as is.

Reply:
at least for ingressControllers, you can view the ingresscontroller object itself since it will have the defaultCertificate field filled out

Reply:
i ended up going with a scheme where we take the certBundle's secret's name and just prepending the clusterName in front

Reply:
I will follow suit in my pull